### PR TITLE
Added remind button on support page and updated remind message

### LINKED ISSUE #74: [Feature Request] - Create remind functionality for support queue
### Pre-requisites:
- [x] I looked through both [open and closed issues](../issues?utf8=✓&q=is%3Aissue) and did not find another request for the same or similar feature.

## Describe the solution you'd like

If the user who requested help doesn't show up after being notified, the sponsor team should be able to send a reminder notification to them via the front end by clicking a `Remind` button which will utilize `/api/supportRequest/remindUser`

Dependent on #175

### PR DESCRIPTION
**Pull request by tatumwashere**
## Pre-Requisites
- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

## Description of Changes

Added a remind button to the UI on the support page. Updated the remindUser message to be mention the voice channel to be joined. Added voiceChannelName parameter to remindUser api and to be used in the remindUser message. Updated support queue tests to reflect parameter changes in remindUser api.
## Related Issues

Closes #74 - Create remind functionality for support queue

`Co-authored-by: Tristen Even <46259649+tristengeven@users.noreply.github.com>`

### CODE REVIEW

File: src/api/supportRequest.ts [hunk 1]
```diff
@@ -243,6 +243,7 @@ supportRequestRoutes.post('/remindUser', async (req, res) => {
       [supportRequest.slackId],
       stringDictionary.remindUser({
         relativeTimeElapsedString,
+        voiceChannelName,
```
Comment (SpencerKaiser) [inline src/api/supportRequest.ts:246]:
```suggestion
```

File: src/api/supportRequest.ts [hunk 2]
```diff
@@ -226,8 +226,8 @@ supportRequestRoutes.patch('/getSpecific', async (req, res) => {
 });
 
 supportRequestRoutes.post('/remindUser', async (req, res) => {
-  const { supportRequestId, relativeTimeElapsedString } = req.body;
-  if (!supportRequestId || !relativeTimeElapsedString) {
+  const { supportRequestId, relativeTimeElapsedString, voiceChannelName } = req.body;
+  if (!supportRequestId || !relativeTimeElapsedString || !voiceChannelName) {
```
Comment (SpencerKaiser) [inline src/api/supportRequest.ts:230]:
```suggestion
  if (!supportRequestId || !relativeTimeElapsedString) {
```

File: src/api/supportRequest.ts [hunk 3]
```diff
@@ -226,8 +226,8 @@ supportRequestRoutes.patch('/getSpecific', async (req, res) => {
 });
 
 supportRequestRoutes.post('/remindUser', async (req, res) => {
-  const { supportRequestId, relativeTimeElapsedString } = req.body;
-  if (!supportRequestId || !relativeTimeElapsedString) {
+  const { supportRequestId, relativeTimeElapsedString, voiceChannelName } = req.body;
```
Comment (SpencerKaiser) [inline src/api/supportRequest.ts:229]:
```suggestion
  const { supportRequestId, relativeTimeElapsedString } = req.body;
```

File: src/components/SupportQueue/SupportQueueItem.tsx [hunk 4]
```diff
@@ -46,6 +46,20 @@ export const SupportQueueItem: React.FC<SupportQueueItemProps> = ({ secret, supp
     await refetch();
   };
 
+  const remindUser = (supportRequestId: number, movedToInProgressAt: string, voiceChannelName: string) => async (): Promise<void> => {
+    const relativeTimeElapsedString = DateTime.fromISO(movedToInProgressAt).toRelative();
+    await fetch('/api/supportRequest/remindUser', {
+      method: 'POST',
+      body: JSON.stringify({ supportRequestId, relativeTimeElapsedString, voiceChannelName }),
```
Comment (SpencerKaiser) [inline src/components/SupportQueue/SupportQueueItem.tsx:53]:
```suggestion
      body: JSON.stringify({ supportRequestId, relativeTimeElapsedString }),
```

File: src/tests/api/supportRequest.test.ts [hunk 5]
```diff
@@ -198,7 +198,7 @@ describe('api/supportRequest', () => {
       const { app } = require('../../app');
       await supertest(app)
         .post('/api/supportRequest/remindUser')
-        .send({ supportRequestId: 1e6, relativeTimeElapsedString: 'some time ago' })
+        .send({ supportRequestId: 1e6, relativeTimeElapsedString: 'some time ago', voiceChannelName: 'voice channel' })
```
Comment (SpencerKaiser) [inline src/tests/api/supportRequest.test.ts:201]:
```suggestion
        .send({ supportRequestId: 1e6, relativeTimeElapsedString: 'some time ago' })
```

File: src/tests/api/supportRequest.test.ts [hunk 6]
```diff
@@ -213,7 +213,7 @@ describe('api/supportRequest', () => {
       const { app } = require('../../app');
       await supertest(app)
         .post('/api/supportRequest/remindUser')
-        .send({ supportRequestId: supportRequest.id, relativeTimeElapsedString: 'some time ago' })
+        .send({ supportRequestId: supportRequest.id, relativeTimeElapsedString: 'some time ago', voiceChannelName: 'voice channel' })
```
Comment (SpencerKaiser) [inline src/tests/api/supportRequest.test.ts:216]:
```suggestion
        .send({ supportRequestId: supportRequest.id, relativeTimeElapsedString: 'some time ago' })
```

File: src/tests/api/supportRequest.test.ts [hunk 7]
```diff
@@ -230,7 +230,7 @@ describe('api/supportRequest', () => {
       const { app } = require('../../app');
       await supertest(app)
         .post('/api/supportRequest/remindUser')
-        .send({ supportRequestId: 1, relativeTimeElapsedString: 'some time ago' })
+        .send({ supportRequestId: 1, relativeTimeElapsedString: 'some time ago', voiceChannelName: 'voice channel' })
```
Comment (SpencerKaiser) [inline src/tests/api/supportRequest.test.ts:233]:
```suggestion
        .send({ supportRequestId: 1, relativeTimeElapsedString: 'some time ago' })
```

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]
We clearly have some tech debt around the Discord/Slack stuff at the moment and I want to clean some stuff up before we try and make the messaging "correct". For now, can you revert the changes to the string and ignore the `voiceChannelName` stuff?

It was also very unclear, but the `voiceChannelName` was really just `supportTicket.name` - one more piece of the tech debt, but just wanted to clarify that we should be retrieving that from the DB, not from the UI's request (which I bet yall would have done had this not been all goofed up).

Other than that this all looks great! Can you post a pic of the UI so @johnkahn can take a look?

**Comment by tatumwashere** [comment]
Here is a screenshot of the remind button in the UI

Screen Shot 2021-03-03 at 4 18 47 PM

**Review by johnkahn — APPROVED** [review]
Awesome 👍

**Review by SpencerKaiser — APPROVED** [review]
Great job! :shipit:

**Pull request merged**